### PR TITLE
IRISSpectrogramSequence

### PR DESCRIPTION
**Pull request by DanRyanIrish**
This PR creates a new class, ```IRISSpectrogramSequence```, which, in addition to the functionality provided to it by its parent class, ```SpectrogramSequence```, supplies functionality to do the following:
* Convert the data and uncertainty in a ```SpectrogramSequence``` between DN and counts.
* Apply and undo exposure time correction to both data and uncertainty.

In addition to introducing this new class, this PR updates the ```IRISSpectrograph``` object in the following ways:
* The use of ```SpectrogramSequence``` is replaced with ```IRISSpectrogramSequence```.
* Uncertainty on the intensity data is calculated when reading in each file during the initialization of the ```IRISSpectrograph``` and then included in the initialization of that file's ```NDCube```.

This PR makes a start in addressing Issue #25.

### CODE REVIEW

**Comment by DanRyanIrish** [comment]
@abit2, @Cadair.  Would you guys be willing to take a look at this PR for me and let me know if you have any feedback?  Cheers.

**Comment by DanRyanIrish** [comment]
Suggestions from @Cadair:
1. Move IRIS units to an ```irispy.units``` module.
2. Add a copy kwarg in the ```.to_``` and exposure time correction methods.

File: irispy/spectrogram.py [hunk 1]
```diff
@@ -72,3 +74,80 @@ def __init__(self, seq):
...
+    def to_counts(self):
+        """Converts data and uncertainty to photon count units."""
+        for i, cube in enumerate(self.data):
+            if "FUV" in cube.meta["detector type"]:
+                DN_unit = iris_tools.DN_UNIT["FUV"]
+            elif cube.meta["detector type"] == "NUV":
+                DN_unit = iris_tools.DN_UNIT["NUV"]
+            else:
+                raise ValueError("Detector type in FITS header not recognized.")
+            if cube.unit == DN_unit or cube.unit == DN_unit/u.s:
```
Comment (Cadair) [inline irispy/spectrogram.py]: I think for units you should use `is`.

File: irispy/spectrogram.py [hunk 2]
```diff
@@ -72,3 +74,80 @@ def __init__(self, seq):
...
         return cu.get_cube_from_sequence(self.seq, item)
+
+class IRISSpectrogramSequence(SpectrogramSequence):
+    """A SpectrogramSequence for IRIS data including additional functionalities."""
+    def to_counts(self):
+        """Converts data and uncertainty to photon count units."""
+        for i, cube in enumerate(self.data):
+            if "FUV" in cube.meta["detector type"]:
+                DN_unit = iris_tools.DN_UNIT["FUV"]
+            elif cube.meta["detector type"] == "NUV":
```
Comment (Cadair) [inline irispy/spectrogram.py]: these two conditionals are inconsitently written

Comment (DanRyanIrish) [reply irispy/spectrogram.py]: Yes, that's intentional.  The FUV types are given as FUV1 and FUV2 but both correspond to the same DN unit.  However, there is only one NUV type so I check it more rigorously.

**Comment by DanRyanIrish** [comment]
@Cadair Any final comments?

**Pull request merged**